src/lapack.jl

```diff
@@ -138,4 +139,112 @@ for (laqps, elty, relty) in ((:slaqps_, :Float32,    :Float32),
+      end
+      lda = stride(A,2)
+      if lda == 0
+          return A, tau, jpvt
+      end # Early exit
+      work  = Vector{$elty}(undef, 1)
+      lwork = BlasInt(-1)
+      iwork = Vector{BlasInt}(undef, n - 1)
+      cmplx = eltype(A)<:Complex
+      if cmplx
+          rwork = Vector{$relty}(undef, 2n)
+      end
+      info = Ref{BlasInt}()
+      k = Ref{BlasInt}()
+      maxc2nrmk = Ref{$relty}()
+      relmaxc2nrmk = Ref{$relty}()
+      for i = 1:2  # first call returns lwork as work[1]
+          if cmplx
+              ccall((@blasfunc($geqp3rk), liblapack), Cvoid,
+                      (
+                        Ref{BlasInt}, #m
+                        Ref{BlasInt}, #n
+                        Ref{BlasInt}, #nrhs
+                        Ref{BlasInt},#kmax
+                        Ref{$relty},  #abstol
+                        Ref{$relty},  #reltol
+                        Ptr{$elty},   #A
+                        Ref{BlasInt}, #lda
+                        Ptr{BlasInt}, #k
+                        Ptr{$relty}, #maxc2nrmk
+                        Ptr{$relty}, #relmaxc2nrmk
+                        Ptr{BlasInt}, #jpvt -> jpiv
+                        Ptr{$elty},   #tau
+                        Ptr{$elty},   #work
+                        Ref{BlasInt}, #lwork
+                        Ptr{$relty}, #rwork
+                        Ptr{BlasInt}, #iwork
+                        Ptr{BlasInt}  #info
+                      ),
+                      m, n, nrhs,
+                      kmax, abstol, reltol,
+                      A, lda,
+                      k, maxc2nrmk, relmaxc2nrmk,
+                      jpvt, tau, work,
+                      lwork, rwork, iwork, info)
+          else
+              #println("running ccall")
+              ccall((@blasfunc($geqp3rk), liblapack), Cvoid,
+                    (
+                      Ref{BlasInt}, #m
+                      Ref{BlasInt}, #n
+                      Ref{BlasInt}, #nrhs 
+                      Ref{BlasInt},#kmax  
+                      Ref{$elty},  #abstol 
+                      Ref{$elty},  #reltol  
+                      Ptr{$elty},   #A
+                      Ref{BlasInt}, #lda
+                      Ref{BlasInt}, #k  
+                      Ptr{$elty}, #maxc2nrmk  
+                      Ptr{$elty}, #relmaxc2nrmk  
+                      Ptr{BlasInt}, #jpvt -> jpiv
+                      Ptr{$elty},   #tau
+                      Ptr{$elty},   #work
+                      Ref{BlasInt}, #lwork
+                      Ptr{BlasInt}, #iwork 
+                      Ptr{BlasInt}  #info
+                    ),
+                    m, n, nrhs,
+                    kmax, abstol, reltol,
+                    A, lda,
+                    k, maxc2nrmk, relmaxc2nrmk,
+                    jpvt, tau, work,
+                    lwork, iwork, info)
+          end
+          chklapackerror(info[])
+          if i == 1
+              lwork = BlasInt(real(work[1]))
```

Comment:
Please review the error messages in the test runs. This test is failing because you are trying to convert a `Float32` to an `Int64`, getting an `InexactError: Int64(1.1435239f-21)`.
https://github.com/JuliaLinearAlgebra/LowRankApprox.jl/actions/runs/12775884079/job/35613281778?pr=61

Reply:
Line 240 is correct and not the cause of the error. See my comment about LAPACK versions. Also this line is doing the exact same thing as line 33.